packages/react-devtools-extensions/src/injectGlobalHook.js

```diff
@@ -96,6 +102,12 @@ if ('text/html' === document.contentType) {
       saveNativeValues +
       detectReact,
   );
+
+  if (
+    typeof window.wrappedJSObject.__REACT_DEVTOOLS_GLOBAL_HOOK__ === 'undefined'
```

Comment:
Seems like this could also happen for other reasons? Such as being in an iframe. That's likely a lot more common. So I worry this would set people on a wrong path.

Reply:
@gaearon I am sure you know more scenarios that can happen here, and I will check what will happen when the website is open in iframe. This is just more-or-less the solution we agreed upon with @bvaughn in the [original issue](https://github.com/facebook/react/issues/17997). But, from your experience, what do you think would be better solution? Obviously, CSP gains more popularity, and I'm pretty sure there will be more and more websites stumbling upon it... Do you thing that comment in README.md is sufficient?

Thanks for your time and attention, I hope we'll find a way to make it better!

Reply:
Sorry for the confusion.

My assumption about the warning was that we would only want to log it if we thought it was likely that the failure was due to CSP, but how to determine this is not something I know much about.

Reply:
@bvaughn no problem, thanks for your help:)

I tried to find any solution to detect CSP preventing inline scripts on the page, but I did not find anything that would work ONLY for CSP problem. There is always some kind of side effect or corner cases. 

What do you think about just adding a couple of lines to the README as I have [here](https://github.com/facebook/react/pull/18107/files#diff-7d04776b03aa659131b7d780d6f9d83fR110)? Is it sufficient to mitigate the problem people complain about?

Reply:
Seems like a reasonable alternative.